### PR TITLE
Flight: Cancel editing on ESCAPE key press

### PR DESCRIPTION
**Pull request by nithindavid**
This will fix Flight app's issue listed in #789

### CODE REVIEW

File: examples/flight/app/js/ui/todo_list.js [hunk 1]
```diff
@@ -34,8 +35,11 @@ define([
 
 		this.edit = function (e, data) {
 			var $todoEl = $(data.el).parents('li');
+			var $inputEl = $todoEl.find('input');
+			var value = $todoEl.find('label').html();
```
Comment (passy) [inline examples/flight/app/js/ui/todo_list.js]: Just wondering, is this a safe thing to do? Could this have implications for HTML that's entered?

Comment (nithindavid) [reply examples/flight/app/js/ui/todo_list.js]: @passy Should I trim the " $todoEl.find('label').html(); " ?

Comment (sindresorhus) [reply examples/flight/app/js/ui/todo_list.js]: `.html()` => `.text()`

File: examples/flight/app/js/ui/todo_list.js [hunk 2]
```diff
@@ -59,9 +63,21 @@ define([
...
+			$todoEl.removeClass('editing');
+			$inputEl.val('');
+
+			this.trigger('uiUpdateRequested');
+		};
+
+		this.requestUpdateOrCancelEvent = function (e, data) {
 			if (e.which === ENTER_KEY) {
 				this.requestUpdate(e, data);
+			} else if (e.which === ESCAPE_KEY) {
```
Comment (sindresorhus) [inline examples/flight/app/js/ui/todo_list.js]: Use `e.key` here instead.

**Comment by nithindavid** [comment]
@sindresorhus added review changes 👍

**Comment by nithindavid** [comment]
@sindresorhus Updated, Thanks 😄

File: examples/flight/app/js/ui/todo_list.js [hunk 3]
```diff
@@ -59,9 +63,21 @@ define([
...
+			var $todoEl = $inputEl.parents('li');
+
+			$todoEl.removeClass('editing');
+			$inputEl.val('');
+
+			this.trigger('uiUpdateRequested');
+		};
+
+		this.requestUpdateOrCancelEvent = function (e, data) {
 			if (e.which === ENTER_KEY) {
```
Comment (sindresorhus) [inline examples/flight/app/js/ui/todo_list.js]: This can't use `e.which` if you use named keys.

File: examples/flight/app/js/ui/todo_list.js [hunk 4]
```diff
@@ -7,7 +7,8 @@ define([
 	'app/utils'
 ], function (defineComponent, todoTmpl, utils) {
 	function todoList() {
-		var ENTER_KEY = 13;
+		var ENTER_KEY = "Enter";
+		var ESCAPE_KEY = "Escape";
```
Comment (sindresorhus) [inline examples/flight/app/js/ui/todo_list.js]: Use single-quotes like the rest of the code.

**Comment by nithindavid** [comment]
@sindresorhus  Sorry for the carelessness 😅 Hope this will be the last of it

**Pull request merged**